### DISCUSSION TITLE
The search function doesn't find "my labels" (tags)

### DISCUSSION THREAD

**Discussion by domalex** [FreshRSS/FreshRSS#6174] [Q&A] [answered]
**Description**

I am testing the new "my labels" feature. The search feature doesn't find feeds with the appropriate "my label" (tag).

**To Reproduce**

1. Create a subscription | Website URL: https://www.tagesanzeiger.ch/, Category: Tageszeitung
2. Create a "my labels" | Name: tagi, Filter actions: inurl:tagesanzeiger.ch/ !Abonnent:in?
4. Enter tagi in the search field

**Expected behavior**

The search result lists all feeds labled with tagi.

**Environment information:**

- Raspberry-Pi 4, 8 GB
- Docker image freshrss/freshrss:edge - currently 1.23.2-dev
- Docker image postgres:16-alpine

**Additional context**

The above is just an example. The bug has been reproduced with other "my labels" (tags).

**Comment by math-GH** [comment]
could you please give a working feed?  https://www.tagesanzeiger.ch/ is not a feed and I cannot find a RSS/Atom feed there

**Comment by domalex** [comment]
This is the feed address to the website above: https://partner-feeds.publishing.tamedia.ch/rss/tagesanzeiger/front. Let me know should you require any further information.

**Comment by Alkarex** [comment]
I am not sure I got the problem/question right, but if you want to search for articles with a given label,

* `label:"Hello World"` (by label name)
* `l:10` (by label ID)

Check the doc https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field

**Reply by domalex** [reply]
In the search field it says: "Search words or #tags"

In the doc it says in regards to "Filtering articles" -> "With the search field" ... "by tag: #tag or #tag+with+whitespace"

As described above, I created a feed and tagged its articles with "tagi". Following I entered in the search field: "#tagi" but got no result at all. The search bar and doc suggest that one can search for words or #tags. However, I was not successful with #tags.

Following @Alkarex I tried to search for label:"tagi" or another label name. Indeed, I find results. However, they do not correspond with the entries of the same label under "My labels" or under "user queries". I expect the same results whatever method is used. I guess this is a bug.

I am not clear whether there is a difference between tags and labels. Is there any or are these just synonyms? Some clarification in the doc would be much appreciated.

**Reply by Alkarex** [reply, marked as answer]
It is indeed confusing, but as you can see in the FreshRSS interface, *user labels* and *article tags* are two different things

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/dec82a85-5c1e-42b3-b214-7fb1c1a06667)

* *user labels* are defined by the user of a FreshRSS instance, by manually or automatically assigning a label to retrieved articles. Those are also the ones that are visible in the *My labels* menu. They are the ones searched by `label:"Hello World"` or `l:10` ([more options in the documentation](https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field))

* *article tags* are defined by the upstream RSS feed, and FreshRSS users have no control over them. They are searched by `#word`.

When you search using the search bar, you need to be sure to:

1) Search from the top view (*Main stream*), otherwise you will only search underneath the category or feed you are in
2) Pick the wanted state (read and/or unread articles) 
![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/028972c5-2a19-4ae6-99c0-cf42aa822414)

**Comment by domalex** [comment]
Thank you for the clarification regarding #tags and labels. My bad that I didn't realize the difference.

It seems that none of my articles have got any #tags. I therefore have never seen the indication of "Article tags" at the end of an article. Only labels are stated.

Following your instructions, I paid attention to search in the relevant stream. This lead to the desired search results.

Thank you for your help.
